infrastructure/async/src/main/java/tech/pegasys/teku/infrastructure/async/SafeFuture.java

```diff
@@ -647,6 +647,24 @@ public SafeFuture<T> whenSuccess(final Runnable action) {
             });
   }
 
+  /**
+   * Returns the future which completes with the same result or exception. The successAction is run
+   * if this future completes successfully and the exceptionAction is run if this future completes
+   * exceptionally
+   */
+  public SafeFuture<T> whenSuccessOrException(
```

Comment:
maybe call it `whenComplete` for simplicity?

Reply:
just removed it for now using separate `whenSuccess` and `whenException`